src/common/src/util/row_serde.rs

```diff
@@ -80,9 +80,28 @@ impl OrderedRowSerde {
             let datum = memcmp_encoding::deserialize_datum(data_type, order, &mut deserializer)?;
             values.push(datum);
         }
+        Ok(values)
+    }
+
+    pub fn deserialize(&self, data: &[u8]) -> memcomparable::Result<OwnedRow> {
+        let values = self.deserialize_impl(data)?;
         Ok(OwnedRow::new(values))
     }
 
+    /// The elements in parameter `output_indices` must be unique, because we take the content of `Datum` for every indices. Taking a `Datum` again would get `None`.
+    pub fn deserialize_by_indices(
```

Comment:
Making this a `safe pub fn` looks unsound to me. 😕 Once it's misused and precondition does not meet, the results will be incorrect.